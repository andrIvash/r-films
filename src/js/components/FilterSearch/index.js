// @flow
import React, { Component } from 'react';

type Props = {
 changeFilter: (elem: HTMLElement) => void,
 selected: string
}

export default class FilterSearch extends Component<Props> {

  _changeFilter = event => this.props.changeFilter(event.target.value);

  render() {
    const { selected } = this.props;
    return (
      <div className='filter-search'>
        <div className='filter-search__title'>Search By</div>
        <div className='btn-group-toggle'>
          <label
            className={selected === 'genres' ?
            'filter-search__label btn btn-secondary active' :
            'filter-search__label btn btn-secondary'}
            htmlFor='genre'
            >
            <input
              autoComplete='off'
              className='filter-search__btn'
              id='genre'
              name='options'
              onChange={this._changeFilter}
              type='radio'
              value='genres'
            /> Genre
          </label>
          <label
            className={selected === 'title' ?
            'filter-search__label btn btn-secondary active' :
            'filter-search__label btn btn-secondary'}
            htmlFor='title'
            >
            <input
              autoComplete='off'
              className='filter-search__btn'
              id='title'
              name='options'
              onChange={this._changeFilter}
              type='radio'
              value='title'
            /> Title
          </label>
        </div>
      </div>
    );
  }
}

