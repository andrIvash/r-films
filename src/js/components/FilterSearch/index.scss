.filter-search {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.filter-search__title {
  font-family: $sign-font;
  font-size: rem(12px);
  color: $color-layout;
  text-transform: uppercase;
  margin-right: rem(10px);
}

.filter-search__label {
  width: rem(70px);
  height: rem(30px);
  font-family: $sign-font;
  font-size: rem(10px);
  margin: 0 rem(5px);
  text-transform: uppercase;
}