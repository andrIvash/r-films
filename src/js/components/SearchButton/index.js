// @flow
import React from 'react';

type Props = {
 handleSearch: (ev: SynteticInputIvent) => void,
}

const SearchButton = (props: Props) => (
  <button
    className='search-button btn btn-lg btn-primary'
    onClick={props.handleSearch}
    type='button'
    > Search
  </button>
);

export default SearchButton;
