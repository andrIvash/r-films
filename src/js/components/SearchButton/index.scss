.search-button {
  width: rem(100px);
  height: rem(40px);
  font-family: $sign-font;
  font-size: rem(14px);
  margin: 0 rem(5px);
  text-transform: uppercase;
}

.search-button {
  background-color: $color-logo;
  border-color: $color-logo;
  &:hover {
    background-color: $color-logo-dark;
    border-color: $color-logo-dark;
  }
  &:active:not(:disabled):not(.disabled) {
    background-color: $color-logo-dark;
  }
}

.header__top {
  .search-button {
    width: rem(70px);
    height: rem(30px);
    font-size: rem(9px);
    background-color: $color-layout;
    color: $color-logo;
    &:hover {
      color: $color-logo-dark;
      background-color: $color-layout-dark;
    }
    &:active:not(:disabled) {
      background-color: $color-logo;
    }

  }
}