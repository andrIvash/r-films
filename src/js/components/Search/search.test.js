import React from 'react';
import { Search } from './index';
import FilterSearch from '../FilterSearch';
import SearchButton from '../SearchButton';

let props = {};

describe('Search', () => {
  beforeEach(() => {
    history.push = jest.fn();
    props = {
      submitSearch: jest.fn(),
      changeFilter: jest.fn().mockReturnValue({searchFilter: 'title'}),
      handleChange: jest.fn().mockReturnValue({searchText: 'text'}),
      clearText: () => jest.fn().mockReturnValue({searchText: ''}),
      history: {push: jest.fn()},
      searchText: '',
    };
  });
  it('should be defined', () => {
    expect(Search).toBeDefined();
  });

  it('should render correctly', () => {
    const wrapper = render(
      <Search {...props} /> );
    expect(wrapper).toMatchSnapshot();
  });

  it('should emit search when click button', () => {
    props.searchText = 'text';
    const wrapper = mount(
      <Search {...props} /> );
    wrapper.find(SearchButton).simulate('click');
    expect(props.submitSearch).toHaveBeenCalledTimes(1);
    expect(props.history.push).toHaveBeenCalledTimes(1);
  });

  it('should change searchText when input data', () => {
    const wrapper = shallow(
      <Search {...props} /> );
    const event = { target: { value: 'test' } };
    wrapper.find('.form-control').simulate('change', event);
    expect(props.handleChange).toHaveBeenCalledTimes(1);
    expect(wrapper.instance().props.searchText.length).not.toEqual('0');
  });

  it('should emit changeFilter when FilterSearch emit', () => {
    const wrapper = mount(
      <Search {...props} /> );
    wrapper.find(FilterSearch).find('#genre').simulate('change');
    expect(props.changeFilter).toHaveBeenCalledTimes(1);
  });

  it('should emit submitSearch when Enter clcked', () => {
    props.searchText = 'text';
    const wrapper = shallow(
      <Search {...props} /> );
    const event = { key: 'Enter' };
    wrapper.find('.form-control').simulate('keypress', event);
    expect(props.submitSearch).toHaveBeenCalledTimes(1);
  });

  it('should not emit submitSearch when not Enter button clcked', () => {
    const wrapper = shallow(
      <Search {...props} /> );
    const event = { key: 'b' };
    wrapper.instance().setState({ searchText: 'text' });
    wrapper.find('.form-control').simulate('keypress', event);
    expect(props.submitSearch).not.toHaveBeenCalledTimes(1);
  });
});
