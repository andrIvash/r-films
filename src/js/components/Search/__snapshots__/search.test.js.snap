// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render correctly 1`] = `
<div
  class="search app__search"
>
  <h1
    class="search__title"
  >
     Find your movie 
  </h1>
  <div
    class="search__input-wrapper input-group mb-3"
  >
    <input
      aria-label="Search"
      class="search__input form-control"
      name="search"
      placeholder="Search phrase..."
      type="text"
      value=""
    />
    <span
      class="search__sign"
    >
      <svg
        fill="#B72E43"
        height="484.5px"
        id="Capa_1"
        version="1.1"
        viewBox="0 0 484.5 484.5"
        width="484.5px"
        x="0px"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <g>
          <g
            id="keyboard-return"
          >
            <polygon
              points="433.5,114.75 433.5,216.75 96.9,216.75 188.7,124.95 153,89.25 0,242.25 153,395.25 188.7,359.55 96.9,267.75  484.5,267.75 484.5,114.75"
            />
          </g>
        </g>
      </svg>
    </span>
  </div>
  <div
    class="search__controls"
  >
    <div
      class="filter-search"
    >
      <div
        class="filter-search__title"
      >
        Search By
      </div>
      <div
        class="btn-group-toggle"
      >
        <label
          class="filter-search__label btn btn-secondary"
          for="genre"
        >
          <input
            autocomplete="off"
            class="filter-search__btn"
            id="genre"
            name="options"
            type="radio"
            value="genres"
          />
           Genre
        </label>
        <label
          class="filter-search__label btn btn-secondary"
          for="title"
        >
          <input
            autocomplete="off"
            class="filter-search__btn"
            id="title"
            name="options"
            type="radio"
            value="title"
          />
           Title
        </label>
      </div>
    </div>
    <button
      class="search-button btn btn-lg btn-primary"
      type="button"
    >
       Search
    </button>
  </div>
</div>
`;
